src/env/from_envp.c

```diff
@@ -11,27 +11,39 @@
 /* ************************************************************************** */
 
 #include <assert.h>
-#include "env.h"
+#include "private.h"
+
+static void		free_new(struct s_env_pair	*new)
+{
+	ft_memdel((void **)new);
+	if (new->key != NULL)
+		ft_strdel(&new->key);
+	if (new->value != NULL)
+		ft_strdel(&new->value);
+}
 
 int				env_from_envp(t_env *env, char **envp)
 {
 	size_t				i;
-	struct s_env_list	*new;
+	struct s_env_pair	*new;
 	size_t				equal_index;
 
 	i = 0;
 	while (envp[i] != '\0')
 	{
-		new = (struct s_env_list *)ft_memalloc(sizeof(*new));
+		new = (struct s_env_pair *)ft_memalloc(sizeof(*new));
 		if (new == NULL)
 			return (-1);
 		assert(ft_strchr(envp[i], '=') != NULL);
```

Comment:
So I did some experimentation by creating two small programs. The parent program calls `execve` on the child program with some env variables that don't contain a `=`. The kernel didn't complain and the child program did get these env variables.

This means that getting a strange env like just the string `FOO` instead of `FOO=1` is possible. I'm not sure what to do in this case. I think just dropping in and possibly printing out a warning would be the most appropriate.

In any case the `assert` needs to go away since we now know that this is something which can happen.

Reply:
I removed the assert. Should I add anything?